packages/rocketchat-lib/client/MessageAction.coffee

```diff
@@ -79,6 +79,33 @@ Meteor.startup ->
 			RocketChat.MessageAction.hideDropDown()
 
 	RocketChat.MessageAction.addButton
+		id: 'reply-message'
+		icon: 'icon-reply'
+		i18nLabel: 'Reply'
+		context: [
+			'message'
+			'message-mobile'
+		]
+		action: (event, instance) ->
+			message = @_arguments[1]
+			input = instance.find('.input-message')
+			url = RocketChat.MessageAction.getPermaLink(message._id)
+			console.log message
+			text = '[ ](' + url + ') @' + message.u.username + ': '
+			if input.value
+				input.value += if input.value.endsWith(' ') then '' else ' '
+			input.value += text
+			input.focus()
+			$(input).keyup()
+			RocketChat.MessageAction.hideDropDown()
+		validation: (message) ->
+			room = RocketChat.models.Rooms.findOne({ _id: message.rid })
```

Comment:
The array of usernames will be removed soon (PR #3918)
Can you replace this check by 
```coffeescript
		validation: (message) ->
			if not RocketChat.models.Subscriptions.findOne({ rid: message.rid })?
				return false

			return true
```


Like we are doing here https://github.com/RocketChat/Rocket.Chat/blob/improvements/more-streams/packages/rocketchat-lib/client/MessageAction.coffee#L183-L187